### PR TITLE
Add recent language version descriptions

### LINKED ISSUE #12188: Updates for C# 8 needed.
We need this page to be updated for C# 8 and Visual Studio 2019

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3d252158-0367-77c8-1816-dbc385c5c40b
* Version Independent ID: 575872e5-6273-f8f6-e3e2-2b911cd9d009
* Content: [The history of C# - C# Guide](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history#feedback)
* Content Source: [docs/csharp/whats-new/csharp-version-history.md](https://github.com/dotnet/docs/blob/staging/docs/csharp/whats-new/csharp-version-history.md)
* Product: **dotnet-csharp**
* GitHub Login: @erikdietrich
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #12188 

The latest C# versions were not described here.

### CODE REVIEW

File: docs/csharp/whats-new/csharp-version-history.md [hunk 1]
```diff
@@ -168,4 +168,86 @@ All of these features offer cool new capabilities for developers and the opportu
...
+- [`private protected` access modifier](./csharp-7-2.md#private-protected-access-modifier)
+  - The `private protected` access modifier enables access for derived classes in the same assembly.
+- [Conditional `ref` expressions](./csharp-7-2.md#conditional-ref-expressions)
+  - The result of a conditional expression (`?:`) can now be a reference.
+
+## C# version 7.3
+
+There are two main themes to the C# 7.3 release. One theme provides features that enable safe code to be as performant as unsafe code. The second theme provides incremental improvements to existing features. In addition, new compiler options were added in this release.
+
+The following new features support the theme of better performance for safe code:
```
Comment (tdykstra) [inline docs/csharp/whats-new/csharp-version-history.md:208]: No links to more info for 7.3?

File: docs/csharp/whats-new/csharp-version-history.md [hunk 2]
```diff
@@ -168,4 +168,86 @@ All of these features offer cool new capabilities for developers and the opportu
...
+- [Default interface methods](./csharp-8.md#default-interface-methods)
+- [Pattern matching enhancements](./csharp-8.md#more-patterns-in-more-places):
+  - [Switch expressions](./csharp-8.md#switch-expressions)
+  - [Property patterns](./csharp-8.md#property-patterns)
+  - [Tuple patterns](./csharp-8.md#tuple-patterns)
+  - [Positional patterns](./csharp-8.md#positional-patterns)
+- [Using declarations](./csharp-8.md#using-declarations)
+- [Static local functions](./csharp-8.md#static-local-functions)
+- [Disposable ref structs](./csharp-8.md#disposable-ref-structs)
+- [Nullable reference types](./csharp-8.md#nullable-reference-types)
```
Comment (tdykstra) [inline docs/csharp/whats-new/csharp-version-history.md:243]: Now that we have an article dedicated to this, should we link directly to that article?

File: docs/csharp/whats-new/csharp-version-history.md [hunk 3]
```diff
@@ -168,4 +168,86 @@ All of these features offer cool new capabilities for developers and the opportu
...
+- [Asynchronous streams](./csharp-8.md#asynchronous-streams)
+- [Indices and ranges](./csharp-8.md#indices-and-ranges)
+- [Null-coalescing assignment](./csharp-8.md#null-coalescing-assignment)
+- [Unmanaged constructed types](./csharp-8.md#unmanaged-constructed-types)
+- [Stackalloc in nested expressions](./csharp-8.md#stackalloc-in-nested-expressions)
+- [Enhancement of interpolated verbatim strings](./csharp-8.md#enhancement-of-interpolated-verbatim-strings)
+
+Default interface members require enhancements in the CLR. Those features were added in the CLR for .NET Core 3.0. Ranges and indexes, and asynchronous streams require new types in the .NET Core 3.0 libraries. Nullable reference types, while implemented in the compiler is much more useful when libraries are annotated to provide semantic information regarding the null state of arguments and return values. Those annotations are being added in the .NET Core libraries.
```
Comment (tdykstra) [inline docs/csharp/whats-new/csharp-version-history.md:251]: Suggestion to add comma after "while implemented in the compiler"

```suggestion
Default interface members require enhancements in the CLR. Those features were added in the CLR for .NET Core 3.0. Ranges and indexes, and asynchronous streams require new types in the .NET Core 3.0 libraries. Nullable reference types, while implemented in the compiler, is much more useful when libraries are annotated to provide semantic information regarding the null state of arguments and return values. Those annotations are being added in the .NET Core libraries.
```

**Review by tdykstra — APPROVED** [review]
Great summaries,  a couple suggestions.

File: docs/csharp/whats-new/csharp-version-history.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/whats-new/csharp-version-history.md:208]: added

**Pull request merged**